topiary-core/src/comments.rs

```diff
@@ -193,48 +207,56 @@ fn find_anchor<'tree>(
     }
 }
 
-pub struct SeparatedInput<'a> {
+struct AnchoredComment<'a>{
+    comment: Node<'a>,
+    commented: Commented,
+}
+
+pub struct SeparatedInput<'a>{
     pub input_tree: Tree,
     pub input_string: String,
-    pub comments: Vec<(&'a Node<'a>, Commented)>,
+    pub comments: Vec<AnchoredComment<'a>>,
 }
 
 // TODO: store comments instead of discarding them
 pub fn extract_comments<'a>(
     tree: Tree,
     input: &str,
     grammar: &Language,
-) -> FormatterResult<(Tree, String)> {
-    let mut comments: Vec<(Node, Commented)> = Vec::new();
+) -> FormatterResult<(Tree, String, Vec<AnchoredComment<'a>>)> {
+    let mut anchors: Vec<AnchoredComment> = Vec::new();
     let mut new_input: String = input.to_string();
     let mut new_tree: Tree = tree;
-    find_comments(new_tree.root_node(), input, &mut comments);
-    comments.sort_by_key(|(node, _)| node.start_byte());
-    comments.reverse();
+    find_comments(new_tree.root_node(), input, &mut anchors)?;
+    anchors.sort_by_key(|AnchoredComment { comment, .. }| comment.start_byte());
     let mut edits: Vec<InputEdit> = Vec::new();
-    for (node, anchor) in comments {
-        match anchor {
-            Commented::CommentedBefore(anchor) => {
-                log::warn!("Commented section precedes comment {:?}:\n{anchor:?}", &node)
-            }
-            Commented::CommentedAfter(anchor) => {
-                log::warn!("Commented section follows comment {:?}:\n{anchor:?}", &node)
-            }
-        }
-        new_input.replace_range((node.start_byte() as usize)..(node.end_byte() as usize), "");
+    // for each (comment, anchor) pair in reverse order, we:
```

Comment:
It might be worth mentioning that this is done in reverse order so that the deletions don't affect one another.